core/deployment/src/main/java/io/quarkus/deployment/steps/CompiledJavaVersionBuildStep.java

```diff
@@ -0,0 +1,55 @@
+package io.quarkus.deployment.steps;
+
+import java.io.DataInputStream;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.concurrent.atomic.AtomicReference;
+
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.pkg.builditem.BuildSystemTargetBuildItem;
+import io.quarkus.deployment.pkg.builditem.CompiledJavaVersionBuildItem;
+
+public class CompiledJavaVersionBuildStep {
+
+    /**
+     * Determines the Java version by looking up the major version of the first successfully parsed
+     * application .class file that is found
+     */
+    @BuildStep
+    public CompiledJavaVersionBuildItem compiledJavaVersion(BuildSystemTargetBuildItem buildSystemTarget) {
```

Comment:
@aloubyansky this is the best way to find the compiled output?

Reply:
That'll do for now, I suppose.